File: 3074.apple-redistribution-into-boxes.py
#
# @lc app=leetcode id=3074 lang=python3
#
# [3074] Apple Redistribution into Boxes
#

# @lc code=start
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        s = 0
        count = 0
        for v in apple:
            s = s + v
        capacity.sort()
        i = len(capacity) - 1
        while(i >= 0):
            s =  s - capacity[i]
            count += 1
            if s<= 0:
                break
            i -= 1
        return count
# @lc code=end

File: 3074.apple-redistribution-into-boxes.java
import java.util.Arrays;
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        // Calculate total apples
        int total = 0;
        for (int a : apple) {
            total += a;
        }
        
        // Sorts in ascending order
        Arrays.sort(capacity);
        
        int cnt = 0;
        // Iterate from the end to pick largest boxes first
        for (int i = capacity.length - 1; i >= 0; i--) {
            total -= capacity[i];
            cnt++;
            if (total <= 0) {
                return cnt;
            }
        }
        return cnt;
    }
	public static void main(String[] args) {
		Solution sol = new Solution();
		int[] apple = {1, 3, 2};
		int[] capacity = {4, 3, 1, 5, 2};
		System.out.println(sol.minimumBoxes(apple, capacity)); // Output: 2
	}
}